api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java

```diff
@@ -1141,7 +1141,19 @@ public ConceptSource getConceptSourceByName(String conceptSourceName) throws DAO
 		criteria.add(Restrictions.eq("source.name", conceptSourceName));
 		return (ConceptSource) criteria.uniqueResult();
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.ConceptDAO#getConceptSourceByUniqueId(java.lang.String)
+	 */
+	public ConceptSource getConceptSourceByUniqueId(String uniqueId) {
+		if (StringUtils.isBlank(uniqueId)) {
+			return null;
```

Comment:
No problem if that's better, I'll do it. Its just what's the difference in asking for the concept source of " " and "2.25.89.1" if they both don't exist